### PR TITLE
BUGFIX: CurlEngine follows chain of Http headers

### LINKED ISSUE #1044: Using a proxy with cURL leads to misbehaving in splitting header and body
### Description

When using a system-wide proxy, TYPO3\Flow\Http fails to correctly split HTTP headers and content. This happens because there is a loop parsing every single response line and stops at the first empty line it finds which usually splits header from content:

File: Response.php, at line 144
`foreach ($lines as $line) {
if ($parsingHeader) {
if (trim($line) === '') {
$parsingHeader = false;
continue;
} ... } ... } ...
`

In case a system wide proxy is set (ex.: env var http_proxy/HTTPS_PROXY in web server or php-fpm config), there are two header blocks in a cURL response while the first block represents the direct answer from proxy and the second one the forwarded block the proxy server received from request target.

### Steps to Reproduce

1. Configure http(s) proxy to use by setting http_proxy and/or HTTPS_PROXY env var in web server config or php-fpm config, depending on what is used
2. Create a HTTP network request by using TYPO3\Flow\Http framework package
3. Call `$response->getContent()` function which then contains the forwarded headers in the first lines

#### Expected behavior

`$response->getContent()` function should only return the request body. The first, usually unnecessary header lines (directly from proxy connection) should be truncated and not parsed as header lines instead of actually needed response headers from target server.

#### Actual behavior

`$response->getContent()` function returns the forwarded HTTP header lines the proxy received from request target before the actual response content starts.

### Affected Versions

Neos: 2.3.14

Flow: 3.3.15

### PR DESCRIPTION
**Pull request by albe**
Under some circumstances, curl will return all the headers of the request
chain, for example when the CURLOPT_FOLLOWLOCATION option is set.
This change will make the CurlEngine follow the chain of response headers
until the final response.

Fixes #1044,  #849, #570

### CODE REVIEW

File: TYPO3.Flow/Classes/TYPO3/Flow/Http/Client/CurlEngine.php [hunk 1]
```diff
@@ -148,9 +148,10 @@ public function sendRequest(Http\Request $request)
         curl_close($curlHandle);
 
         $response = Http\Response::createFromRaw($curlResult);
-        if ($response->getStatusCode() === 100) {
+        while (substr($response->getContent(), 0, 5) === 'HTTP/' || $response->getStatusCode() === 100) {
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Http/Client/CurlEngine.php]: Actually not sure if the `100` code check is still needed with this, but I kept it in place just to be sure.

**Comment by kdambekalns** [comment]
@thelalaman

File: TYPO3.Flow/Classes/TYPO3/Flow/Http/Client/CurlEngine.php (hunk 1 above)
Comment (bwaidelich) [inline TYPO3.Flow/Classes/TYPO3/Flow/Http/Client/CurlEngine.php]: Is that safe to check the response _content_ to start with "HTTP/"?
Seems a bit "magic" to me. Wouldn't the `$curlResult` contain more details about this so we can catch that case more "safely"?

Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Http/Client/CurlEngine.php]: I asked that myself too, but there is really no other information in `$curlResult`. It just appends the response headers after each other. We could only apply some more checks on the format of the content to make more sure it is indeed a HTTP Response, like checking for exact line delimiters (CRLF), exact status code match (3 digits) and possible headers in `identifier: value(;value)*?` format and followed by an empty line.

**Comment by bwaidelich** [comment]
@kdambekalns did your comment have any deeper meening or was that a happy accident? ;)

**Comment by kdambekalns** [comment]
It was meant to draw attention to this PR, since I know @thelalaman had that issue and his colleague opened the latest of the issues this should fix.

File: TYPO3.Flow/Classes/TYPO3/Flow/Http/Client/CurlEngine.php (hunk 1 above)
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Http/Client/CurlEngine.php]: probably `Response::createFromRaw` should just be more strict on those checks itself and then we catch any error and break the loop.

**Comment by albe** [comment]
This should prevent most cases that somehow respond with seemingly HTTP response header content by making the parsing more strict and CurlEngine catching and stopping on eceptions.

**Comment by albe** [comment]
Yeah, that's a good point for things to go over generally in the tests.

**Pull request merged**